benchmarks/naslib/problem.py

```diff
@@ -0,0 +1,155 @@
+import ast
+import os
+import sys
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from kurobako import problem
+
+
+sys.stdout = open(os.devnull, "w")  # Suppress output
+
+
+import naslib.search_spaces as ss  # NOQA
+from naslib.search_spaces.core.graph import Graph  # NOQA
+from naslib.search_spaces.core.query_metrics import Metric  # NOQA
+import naslib.search_spaces.nasbench201.graph as nasbench201_graph  # NOQA
+import naslib.search_spaces.transbench101.graph as transbench101_graph  # NOQA
+from naslib.utils import get_dataset_api  # NOQA
+
+
+class NASLibProblemFactory(problem.ProblemFactory):
+    def __init__(self, search_space: Graph, **config: Dict[str, Any]) -> None:
+        self._search_space = search_space
+        self._config = config
+
+    def specification(self) -> problem.ProblemSpec:
+        if isinstance(self._search_space, ss.TransBench101SearchSpaceMicro):
+            params = [
+                problem.Var(f"x{i}", problem.CategoricalRange(transbench101_graph.OP_NAMES))
+                for i in range(6)
+            ]
+            self._converter = lambda x: [int(z) for z in x]
+        # elif isinstance(self._search_space, ss.NATSBenchSizeSearchSpace):
+        #     params = [problem.Var(f"x{i}", problem.DiscreteRange(1, 8)) for i in range(5)]
+        #     self._converter = lambda x: [z * 8 for z in x]
+        elif isinstance(self._search_space, ss.NasBench201SearchSpace):
+            params = [
+                problem.Var(f"x{i}", problem.CategoricalRange(nasbench201_graph.OP_NAMES))
+                for i in range(6)
+            ]
+            self._converter = lambda x: [int(z) for z in x]
+        else:
+            raise NotImplementedError(f"{self._search_space} is not supported")
+        return problem.ProblemSpec(
+            name=f"{self._search_space}",
+            params=params,
+            values=[problem.Var("value")],
+        )
+
+    def create_problem(self, seed: int) -> problem.Problem:
+        # Pylance wrongly reports type error here
+        return NASLibProblem(self._search_space, self._converter, **self._config)  # type: ignore
+
+
+class NASLibProblem(problem.Problem):
+    def __init__(
+        self, search_space: Graph, converter: Any, direction: str, **config: Dict[str, Any]
+    ) -> None:
+        """"""
+        # params
+        # metric      : Metric to query for
+        # dataset     : Dataset to query for
+        # epoch       : If specified, returns the metric of the arch at that epoch of training
+        # full_lc     : If true, returns the curve of the given metric in all epochs
+        # dataset_api : API to use for querying metrics
+        # hp          : Number of epochs the model was trained for. Value is in {1, 12, 90}
+        # is_random   : When True, the performance of a random architecture will be returned
+        #               When False, the performanceo of all trials will be averaged.
```

Comment:
Do these lines explain the key and value of `config`? If so, I think it is good to explicitly define those arguments .

Reply:
The keys and values of `config` is directly passed to `naslib`, and `naslib` could take different configurations for different benchmarks. This is why I chose to pass a Dict rather than explicitly writing them out as arguments.

Reply:
This comment is actually extracted from one `search_space` in `naslib` and may not be common to all `search_space`s.